shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1340,7 +1389,107 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, var ne
         RecordType.mk(fs)(prov)
       } else TupleType(fields.reverseIterator.mapValues(_.toUpper(noProv)))(prov)
   }
-  
+
+  def getNewVarName(prefix: String, nonValidVars: Set[Var])(implicit raise: Raise): String = {
```

Comment:
The implicit parameter `raise` is never used.
```suggestion
  def getNewVarName(prefix: String, nonValidVars: Set[Var]): String = {
```